barman/utils.py

```diff
@@ -866,3 +868,37 @@ def get_backup_id_using_shortcut(server, shortcut, BackupInfo):
         backup_id = shortcut
 
     return backup_id
+
+
+def lock_files_cleanup(lock_dir, lock_directory_cleanup):
+    """
+    Get all the lock files in the lock directory
+    and try to acquire every single one.
+    If the file is not locked, remove it.
+
+    This method is part of cron and should help
+    keeping clean the lockfile directory.
+    """
+    if not lock_directory_cleanup:
+        # Auto cleanup of lockfile directory disabled.
+        # Log for debug only and return
+        _logger.debug("Auto-cleanup of '%s' directory disabled" % lock_dir)
+        return
+    _logger.info("Cleaning up lockfiles directory.")
+    for filename in glob("%s/*.lock" % lock_dir):
```

Comment:
This glob expression doesn't match any dot-prefixed lock files - I think `"%s/.*.lock" % lock_dir` would work.

Reply:
right, fixed it and also used `os.path.join` to create the path instead than the plain substitution